src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -241,7 +255,19 @@ string EncodeUrl (Uri url)
 			};
 			while (true) {
 				URL java_url = new URL (EncodeUrl (redirectState.NewUrl));
-				URLConnection java_connection = java_url.OpenConnection ();
+				URLConnection java_connection;
+				if (UseProxy)
+					java_connection = java_url.OpenConnection ();
+				else
+					java_connection = java_url.OpenConnection (Java.Net.Proxy.NoProxy);
+
+				var httpsConnection = java_connection as HttpsURLConnection;
+				if (httpsConnection != null) {
+					IHostnameVerifier hnv = GetSSLHostnameVerifier (httpsConnection);
+					if (hnv != null)
+						httpsConnection.HostnameVerifier = hnv;
```

Comment:
More that it's a property with `unknown` default value and we don't want to disturb the default state unless we're 100% sure what the effect would be